pep-0704.rst

```diff
@@ -0,0 +1,137 @@
+
+Installers MAY choose to implement this default behaviour on any Python versions, but MUST implement it on Python 3.13 or newer.
+
+
+Backwards Compatibility
+=======================
+
+This PEP is backwards incompatible with workflows where users are using installers outside of virtual environments. Such users will be prompted with an error message and will need to either:
+
+- explicitly opt-in to running the installer outside of a virtual environment, or
+- create and use a virtual environment
+
+Users who are already using virtual environments will not be affected by this change.
+
+Workflow tools (which manage virtual environments for the user, under the hood) should be unaffected, since they should already be using the virtual environment for running the installer.
+
+
+Security Implications
+=====================
+
+This PEP does not introduce any new security implications.
+
+
+How to Teach This
+=================
+
+This PEP does require that new users create and use a virtual environment to get started with using Python packages. This is however a best practice, as demonstrated by the section on "basics of how to install Python packages" in the Python Packaging User Guide, which explains how/what virtual environments are before discussing using ``pip``.[1]_
+
+
+Reference Implementation
+========================
+
+There is no reference implementation for this PEP, however the proposed behaviour is already implemented in ``pip`` and can be achieved by setting ``PIP_REQUIRE_VENV`` environment variable to ``1`` (not setting it would result in the proposed opt-in behaviour).
+
+Rejected Ideas
+==============
+
+Do not specify a name for the virtual environment directory
+-----------------------------------------------------------
+
+Using a consistent name for the virtual environment directory is important for a few reasons:
+
+1. It makes it easier for users to find the virtual environment directory, and to activate it.
+2. It removes a decision point for new users, since they do not need to decide on a name for the virtual environment directory.
+3. It creates a clear convention within the ecosystem, which makes it easier for users to find documentation.
+4. It ensures consistency across different tools, so that differences in the error message of different tools do not confuse users.
+
+Use a different name for the virtual environment directory
+----------------------------------------------------------
+
+Functionally, the directory name does not matter much as long as there is a single consistent suggestion.
+
+The name ``.venv`` was picked since it:
+
+1. does not conflict with any valid Python import name
+2. does not conflict ``venv`` module in the standard library
+3. has pre-existing usage in the Python community
+4. has support for auto-detection in common text editors
+5. can be typed without modifier keys on common keyboard layouts
```

Comment:
```suggestion
* does not conflict with any valid Python import name
* does not conflict ``venv`` module in the standard library
* has pre-existing usage in the Python community
* has support for auto-detection in common text editors
* can be typed without modifier keys on common keyboard layouts
```

Same here

Reply:
I prefer the numbered list here, since this is an ordered list to me.

Reply:
Fair enough